overlord/devicestate/devicestate.go

```diff
@@ -464,50 +475,15 @@ func remodelKernelOrBaseTasks(ctx context.Context, st *state.State, ms modelSnap
 					// the update is not a simple
```

Comment:
does snapstateUpdateWithDeviceContext do the right thing w/respect to generating a update-gadget-assets task for kernel snap refreshes too (because DTBs are evil and hate us)? I assume so?

Reply:
The corner case should be handled now.